File: core/src/main/java/com/mcreater/amclcore/command/ansi/ColorTheme.java
package com.mcreater.amclcore.command.ansi;

import org.fusesource.jansi.Ansi;

import java.awt.*;

public interface ColorTheme {
    Color getFatal();

    Color getError();

    Color getWarning();

    Color getInfo();

    Color getDebug();

    default Ansi.Consumer apply(Color color) {
        return c -> {
            c.fgRgb(color.getRed(), color.getGreen(), color.getBlue());
        };
    }
}

File: core/src/main/java/com/mcreater/amclcore/command/ansi/ColorTheme.kt
package com.mcreater.amclcore.command.ansi

import org.fusesource.jansi.Ansi


interface ColorTheme {
    fun applyFatal(): Ansi.Consumer?
    fun applyError(): Ansi.Consumer?
    fun applyWarning(): Ansi.Consumer?
    fun applyInfo(): Ansi.Consumer?
    fun applyDebug(): Ansi.Consumer?
    fun applyTrace(): Ansi.Consumer?
}
